src/main/scala/is/hail/methods/IBD.scala

```diff
@@ -348,23 +349,44 @@ object IBD {
 
     val mafSymbolTable = Map("va" -> (0, vds.rowType))
     val mafEc = EvalContext(mafSymbolTable)
-    val computeMafThunk = RegressionUtils.parseFloat64Expr(computeMafExpr, mafEc)
-    val rowType = vds.rvRowType
+    val mafAst = Parser.parseToAST(computeMafExpr, mafEc)
 
     val rowKeysF = vds.rowKeysF
     val localRowType = vds.rowType
 
-    { (rv: RegionValue) =>
-      val row = new UnsafeRow(localRowType, rv)
-      mafEc.setAll(row)
-      val maf = computeMafThunk()
+    val computeMafFromRV = mafAst.toIR() match {
```

Comment:
it could be a lot easier to put this in Python instead (optionally annotate rows) and just do a field lookup in Scala. I haven't looked at IBD in a long time though

Reply:
do you mind if I PR against your PR?

The main thing I'm worried about is inconsistency. In our Python interface, it looks like expressions are treated the same everywhere, but as long as we have methods that do their own expr compilation, that won't be true. Here if a user supplies an expression that contains an aggregation or references a global field, they'll get an error. Instead of documenting the difference or worrying about consistency, I think it'll be easier to just reuse our Python machinery to produce the simplest possible thing going into Scala, consequently tightening up our Scala code and making any future lifts (like to C++) that much easier.

Reply:
Thanks.  Sounds good to move the expression-handling into Python eventually, to guarantee consistency.